Improve clangify.sh

ebbit1q opened the pull request:
Separated from #3433 merge this branch first!

tooomm commented:
Can you please provide a proper description and explanation?
Somebody seeing this isolated in the history needs to understand what changed and why you did it this way. 👍

ebbit1q commented:
benchmark:
```
$ git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
$ sed -i 's/^ \+//g' common/*.cpp
$ time ./clangify.sh
Successfully formatted following files:
common/decklist.cpp
common/featureset.cpp
common/get_pb_extension.cpp
common/rng_abstract.cpp
common/rng_sfmt.cpp
common/server.cpp
common/server_abstractuserinterface.cpp
common/server_arrow.cpp
common/server_card.cpp
common/server_cardzone.cpp
common/server_counter.cpp
common/server_game.cpp
common/server_player.cpp
common/server_protocolhandler.cpp
common/server_remoteuserinterface.cpp
common/server_response_containers.cpp
common/server_room.cpp
common/serverinfo_user_container.cpp

real    0m1.724s
user    0m0.297s
sys     0m0.953s
$ git checkout clangify++
Switched to branch 'clangify++'
Your branch is up to date with 'ebbit1q/clangify++'.
$ sed -i 's/^ \+//g' common/*.cpp
$ time ./clangify.sh

real    0m0.539s
user    0m0.250s
sys     0m0.156s
```

ZeldaZach commented:
Seems like you're overly complicating what was a simple script... but if it works and is well doc'd I'm fine with it

ebbit1q commented:
I actually enjoyed learning some bash.

ctrlaltca commented on clangify.sh:
```diff
@@ -1,22 +1,175 @@
+  diff)
+    declare -i code=0
+    for name in ${names[@]}; do
+      if ! $cf_cmd "$name" | diff "$name" - -p --color=always; then
```
`--color` is available only on GNU diff, it will fail on mac/bsd. Maybe you can use git diff here, too?

ebbit1q replied:
color is now a separate option
